helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1165,7 +1164,11 @@ public void run() {
               + _processorName);
       while (!isInterrupted()) {
         try {
-          handleEvent(_eventBlockingQueue.take(), _cache);
+          ClusterEvent newClusterEvent = _eventBlockingQueue.take();
+          // rename the current processor's thread name to include event id
+          String currentThreadName = this.getName().split("_")[0];
```

Comment:
The thread name shall be fixed, You can just regenerate the name according to our name convention.

Reply:
Done